### PR TITLE
Fix ResourceWarnings from unclosed files #11091

### LINKED ISSUE #11091: Fix ResourceWarnings from unclosed files
Running the tests with `ResourceWarning` enabled flags many instances of unclosed file handles, which is a performance consideration.

We should rewite these `open` calls as `with open(...`:

### Reproducer
```sh
python3 -Wall::ResourceWarning manage.py test tests --settings=tests.test_settings --verbosity=2
```

### Results
Items like:
```py
test_csv_export (tests.exporter.resource_export_tests.BusinessDataExportTests.test_csv_export) ... /Users/jwalls/prj/arches/arches/app/utils/data_management/resources/importer.py:111: ResourceWarning: unclosed file <_io.TextIOWrapper name='tests/fixtures/data/csv/resource_export_test.mapping' mode='r' encoding='UTF-8'>
  self.mapping = json.load(open(path, "r"))
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/Users/jwalls/prj/arches/tests/exporter/resource_export_tests.py:73: ResourceWarning: unclosed file <_io.TextIOWrapper name='tests/fixtures/data/csv/resource_export_test.csv' mode='r' encoding='utf-8'>
  BusinessDataImporter(
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/Users/jwalls/prj/arches/arches/app/utils/data_management/resources/formats/csvfile.py:73: ResourceWarning: unclosed file <_io.TextIOWrapper name='tests/fixtures/data/csv/resource_export_test.mapping' mode='r' encoding='UTF-8'>
  resource_export_configs = json.load(open(configs, "r"))
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/Users/jwalls/prj/arches/tests/exporter/resource_export_tests.py:87: ResourceWarning: unclosed file <_io.TextIOWrapper name='tests/fixtures/data/csv/resource_export_test.csv' mode='r' encoding='utf-8'>
  csv_input = list(
ResourceWarning: Enable tracemalloc to get the object allocation traceback
...
```

### PR DESCRIPTION
**Pull request by HarisMughal**
### Types of changes

-   [X] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change

Replace `open` calls with `open(...)`  to avoid `ResourceWarning`

### Issues Solved

Closes #11091

### Checklist

-   I targeted one of these branches:
    - [X] dev/7.6.x (under development): features, bugfixes not covered below
    - [ ] dev/7.5.x (main support): regressions, crashing bugs, security issues, major bugs in new features
    - [ ] dev/6.2.x (extended support): major security issues, data loss issues
-   [ ] I added a changelog in arches/releases
-   [ ] I submitted a PR to arches-docs (if appropriate)
-   [ ] Unit tests pass locally with my changes
-   [ ] I added tests that prove my fix is effective or that my feature works
-   [ ] My test fails on the target branch

#### Ticket Background
*   Sponsored by: 
*   Found by: @jacobtylerwalls 
*   Tested by: @HarisMughal 
*   Designed by: @

### CODE REVIEW

**Comment by HarisMughal** [comment]
Apologies for any mistakes, this is my first contribution.
Let me know if I need to change anything

File: tests/exporter/resource_export_tests.py [hunk 1]
```diff
@@ -84,15 +84,16 @@ def test_csv_export(self):
...
+            csv_input = list(
+                csv.DictReader(
+                    f,
+                    restkey="ADDITIONAL",
+                    restval="MISSING",
+                )
+            )[0]
 
-        self.assertDictEqual(dict(csv_input), dict(csv_output))
+            self.assertDictEqual(dict(csv_input), dict(csv_output))
```
Comment (jacobtylerwalls) [inline tests/exporter/resource_export_tests.py:96]: OK to leave this unindented

File: arches/app/utils/data_management/resources/importer.py [hunk 2]
```diff
@@ -127,7 +129,8 @@ def __init__(self, file=None, mapping_file=None, relations_file=None):
                             if "business_data" in list(archesfile.keys()):
                                 self.business_data = archesfile["business_data"]
                     elif self.file_format == "csv":
-                        data = csv.DictReader(open(file[0], encoding="utf-8"))
+                        with open(file[0], encoding="utf-8") as f:
+                            data = csv.DictReader(f)
                         self.business_data = list(data)
```
Comment (jacobtylerwalls) [inline arches/app/utils/data_management/resources/importer.py:134]: This also needs indenting, see test failure

File: arches/management/commands/packages.py [hunk 3]
```diff
@@ -918,7 +919,8 @@ def load_business_data(package_dir, prevent_indexing):
             config_paths = glob.glob(os.path.join(package_dir, "package_config.json"))
             configs = {}
             if len(config_paths) > 0:
-                configs = json.load(open(config_paths[0]))
+                with open(config_paths[0]) as f:
+                    configs = json.load()
```
Comment (jacobtylerwalls) [inline arches/management/commands/packages.py:923]:
```suggestion
                    configs = json.load(f)
```

**Review by jacobtylerwalls — CHANGES_REQUESTED** [review]
Thanks for this contribution! I pushed minor formatting edits.

(For your next contribution, you can install the dev requirements under /install/requirements_dev.txt and then run `pre-commit install` to get the formatting hook to run on commit.)

Then, that exposed a few failing unit tests. Once fixed, this should be mergeable. Thanks again.

**Review by jacobtylerwalls — APPROVED** [review]
Thanks for the contribution!

**Comment by HarisMughal** [comment]
IDK why autopep8 did not work on my VScode so I had to manually run it. I ran all lib install in req_dev.txt but was unaware of hooks. I will be careful next time

Thank you for letting me contribute! @jacobtylerwalls

**Pull request merged**